samza-core/src/test/java/org/apache/samza/storage/MockStorageEngine.java

```diff
@@ -35,6 +35,8 @@
   public static String storeName;
   public static File storeDir;
   public static SystemStreamPartition ssp;
+
+  // Thread-safe list is required because the list is shared across StorageEngine instances
   public static ArrayList<IncomingMessageEnvelope> incomingMessageEnvelopes = new ArrayList<IncomingMessageEnvelope>();
```

Comment:
Looks like there's only a comment added as a part of this PR. Is your fix not updated in the PR?